B/new tri wnear farm

memacius32 opened the pull request:
Add TRI<>WNEAR MCV2 farm

<img width="390" alt="Screen Shot 2022-09-29 at 17 28 28" src="https://user-images.githubusercontent.com/96993065/193073978-2ec2c462-876d-4087-b03f-e2ebe27edf61.png">

<img width="773" alt="Screen Shot 2022-09-29 at 17 27 59" src="https://user-images.githubusercontent.com/96993065/193073843-d2c01840-cd00-4a5f-9b36-97e0ee85cf1a.png">

specialist0x commented:
What changes need to be made to the existing `TRI-wNEAR` farm?

<img width="931" alt="image" src="https://user-images.githubusercontent.com/94581898/193081485-3f45b6ce-dbfc-408a-86a2-9ab5b1d2e1bf.png">

memacius32 commented:
- Remove isPeriodFInished hardcoded as false in some hooks.
- Add isPeriodFinished as ** true ** to existing mcv1 tri-wnear farm (disables deposit button)

 <img width="791" alt="Screen Shot 2022-09-29 at 18 29 20" src="https://user-images.githubusercontent.com/96993065/193087769-01909d76-b7c3-4c43-8dbe-f51494f62ae8.png">

specialist0x commented on src/state/stake/useFarmsAPI.ts:
```diff
@@ -49,7 +50,7 @@ export function useFarmsAPI(): StakingTriFarms[] {
       lpAddress,
       rewarderAddress,
       tokens,
-      isPeriodFinished: false,
```
I don't think this is needed

specialist0x commented on src/state/stake/user-farms.ts:
```diff
@@ -66,7 +66,7 @@ export function useSingleFarm(version: number): StakingTri {
     return {
       ...activeFarms[version],
       tokens: tokens!,
-      isPeriodFinished: false,
```
This change isn't needed

specialist0x commented on src/state/stake/user-stable-farms.ts:
```diff
@@ -58,7 +58,7 @@ export function useSingleStableFarm(version: number, stableSwapPoolName: StableS
 
     return {
       ...activeFarms[version],
-      isPeriodFinished: false,
```
This change isn't needed

memacius32 commented:
- Removed unnecesary destructure of useFarmsApi, now uses spread operator as the other two hooks
- Removed isPeriodFinished unnecesary explicit return in userSingleFarm and useSingleStableFarm

specialist0x commented on src/state/stake/useFarmsAPI.ts:
```diff
@@ -21,52 +21,23 @@ export function useFarmsAPI(): StakingTriFarms[] {
-      rewarderAddress,
-      tokens,
-      isPeriodFinished,
+      ...activeFarms[index],
```
👏🏽👏🏽👏🏽
